solc/CommandLineInterface.cpp

```diff
@@ -1097,6 +1097,21 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
```

Comment:
Note that this function processes a lot of option combinations. Some of them do not actually result in full compilation being performed. For example the `--stop-after` I already mentioned (though I'm not 100% certain that one actually enters here) but also `--hashes`, which is why command-line tests fail. You might need to add a check that `CompilerStack` state is `CompilationSuccessful` but you should also make sure this does not exclude cases where we actually want the message to show up.